modules/compute/gke-node-pool/reservation_definitions.tf

```diff
@@ -68,7 +68,8 @@ locals {
   # Compare two maps by counting the keys that mismatch.
   # Know that in map comparison the order of keys does not matter. That is {NVME: x, SCSI: y} and {SCSI: y, NVME: x} are equal
   # As of this writing, there is only one reservation supported by the Node Pool API. So, directly accessing it from the list
-  specific_reservation_requirement_violations = length(local.reservation_vm_properties) == 0 ? [] : [for k, v in local.nodepool_vm_properties : k if v != local.reservation_vm_properties[0][k]]
+  specific_reservation_requirement_violations = local.has_gpu ? (length(local.reservation_vm_properties) == 0 ? [] : [for k, v in local.nodepool_vm_properties : k if v != local.reservation_vm_properties[0][k]]) : []
```

Comment:
There is no `instance_properties` information for the TPU reservation I am using. So, this always fails. Need to fix this.